.github/workflows/release-tarball.yml

```diff
@@ -0,0 +1,35 @@
+name: Release Tarball
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  upload-release-tarball:
+    name: upload complete source code for release
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout tagged release recursively incl. submodules
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: set environment variables
+      run: |
+        NAME="${{ github.repository }}"
```

Comment:
The naming of this variable is quite general. This environment variable holds the name of the tarball - while `NAME` is not wrong, it is more descriptive and less likely to collide with other variable names if it's named something like `RELEASE_TARBALL_NAME`.

Reply:
Obvious at hindsight, thanks, will update in a bit.

Reply:
Renamed (your suggestions) and rebased.